src/aws-iac-mcp-server/awslabs/aws_iac_mcp_server/server.py

```diff
@@ -333,58 +357,6 @@ async def search_cdk_documentation(query: str) -> str:
     return sanitize_tool_response(json.dumps(response_dict))
 
 
-@mcp.tool()
-async def read_iac_documentation_page(
-    url: str,
-    starting_index: int = 0,
-) -> str:
-    """Fetch and convert any Infrastructure as Code (CDK or CloudFormation) documentation page to markdown format.
-
-    ## Usage
-
-    This tool retrieves the complete content of a specific CDK or CloudFormation documentation page. Use it when you need detailed information from a particular document rather than the limited context from the search results.
-
-    ## When to Use
-
-    After using a search tool, use this tool to fetch the complete content of any relevant page in the search results.
-
-    ## Supported Document Types
-
-    - API reference pages
-    - Guide and tutorial pages
-
-    ## Pagination
-
-    For long documents, use the starting_index parameter to read content in chunks. Always use the same URL for all pagination calls to maintain consistency.
-
-    ## Result Interpretation
-
-    Returns JSON with:
-    - knowledge_response: Details of the response
-      - results: Array with single result containing:
-        - rank: Always 1 for document reads
-        - title: Document title or filename
-        - url: Source URL of the document
-        - context: Full or paginated document content
-    - next_step_guidance: If present, suggested next actions to take for answering user query
-
-    For pagination, use starting_index from previous response to continue reading.
-
-    Args:
-        url: URL from search results to read the full page content
-        starting_index: Starting character index for pagination (default: 0)
-
-    Returns:
-        List of search results with URLs, titles, and context snippets
-    """
-    result = await read_iac_documentation_page_tool(url, starting_index)
-
-    # Convert dataclass to dict for JSON serialization
-    response_dict = asdict(result)
-
-    return sanitize_tool_response(json.dumps(response_dict))
```

Comment:
The sanitisation is not being applied to the proxied read tool, since it is a pure proxy of the remote AWS knowledge MCP tool i.e. the response will be exactly what the remote tool returns. Applying sanitisation would require first parsing the ToolResult into a string, needing us to make assumptions about the structure of the response from the remote tool - the very thing we're trying to avoid by making this change (decoupling from the remote tool request/response structures).